src/algebra/tsze.lean

```diff
@@ -0,0 +1,273 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import algebra.algebra.basic
+
+/-!
+# Trivial Square-Zero Extension
+
+Given a module `M` over a ring `R`, the trivial square-zero extension of `M` over `R` is defined
+to be the `R`-algebra `R ⊕ M` with multiplication given by
+`(r₁ + m₁) * (r₂ + m₂) = r₁ r₂ + r₁ m₂ + r₂ m₁`.
+
+It is a square-zero extension because `M^2 = 0`.
+-/
+
+universes u v w
+
+/--
+"Trivial Square-Zero Extension".
+
+Given a module `M` over a ring `R`, the trivial square-zero extension of `M` over `R` is defined
+to be the `R`-algebra `R ⊕ M` with multiplication given by
```

Comment:
Based on the implementation, shouldn't this be a `prod` type not a `sum` type?
```suggestion
to be the `R`-algebra `R × M` with multiplication given by
```

Reply:
This is using the mathematical direct sum notation. It conflicts with Lean notation, so it's probably a good idea to disambiguate.